erigon-lib/kv/kv_interface.go

```diff
@@ -464,6 +464,9 @@ type TemporalTx interface {
 	Tx
 	TemporalGetter
 
+	// return the earliest known txnum in state history (excluding commitment and receipt history)
+	StateHistoryStartFrom() uint64
```

Comment:
`Tx`/`TemporalTx` interface - is low-level interface. it must be business-logic agnostic. 
"State" - it's something from Ethereum business-dictionary. 
`TemporalTx` must not know - which domains are exist (or may be created in future). It also doesn't know what stored inside each domain (for `TemporalTx` - it's just some keys/values/timestamps).

Reply:
Fair point, I'll check if I can make it more general, i.e. HistoryStartFrom()